src/credentials/tests/TestDacOnlyPartialAttestationVerifier.cpp

```diff
@@ -0,0 +1,283 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <pw_unit_test/framework.h>
+
+#include <app/tests/suites/credentials/TestHarnessDACProvider.h>
+#include <controller/CommissioneeDeviceProxy.h>
+#include <credentials/DeviceAttestationConstructor.h>
+#include <credentials/attestation_verifier/DacOnlyPartialAttestationVerifier.h>
+#include <credentials/attestation_verifier/DeviceAttestationVerifier.h>
+#include <credentials/tests/CHIPAttCert_test_vectors.h>
+#include <credentials/tests/CHIPCert_unit_test_vectors.h>
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/CHIPVendorIdentifiers.hpp>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/Span.h>
+
+using namespace chip;
+using namespace chip::Credentials;
+using namespace chip::TestCerts;
+using chip::Credentials::Examples::TestHarnessDACProvider;
+
+namespace {
+
+// Callback function to capture attestation verification results
+static void OnAttestationInformationVerificationCallback(void * context, const DeviceAttestationVerifier::AttestationInfo & info,
+                                                         AttestationVerificationResult result)
+{
+    AttestationVerificationResult * pResult = reinterpret_cast<AttestationVerificationResult *>(context);
+    *pResult                                = result;
+}
+
+} // namespace
+
+struct TestDacOnlyPartialAttestationVerifier : public ::testing::Test
+{
+    static void SetUpTestSuite() { ASSERT_EQ(chip::Platform::MemoryInit(), CHIP_NO_ERROR); }
+    static void TearDownTestSuite() { chip::Platform::MemoryShutdown(); }
+
+    void SetUp() override { attestationResult = AttestationVerificationResult::kSuccess; }
+
+    // Create attestation verification result variable
+    // This will be used to capture the result of the verification callback
+    PartialDACVerifier verifier;
+    AttestationVerificationResult attestationResult = AttestationVerificationResult::kSuccess;
+    Callback::Callback<DeviceAttestationVerifier::OnAttestationInformationVerification> attestationCallback{
+        OnAttestationInformationVerificationCallback, &attestationResult
+    };
+};
+
+// Test verifier behavior with empty parameters
+TEST_F(TestDacOnlyPartialAttestationVerifier, TestWithInvalidParameters)
+{
+    // Create AttestationInfo with empty buffers
+    DeviceAttestationVerifier::AttestationInfo invalidInfo(ByteSpan(),                    // empty attestationElements
```

Comment:
nit: consider something like `ByteSpan() /* attestationElements */` or `/* attestationElements = */ BytesSpan()` since it is obvious that these are empty.